stack/building-with-pip.rst

```diff
@@ -0,0 +1,241 @@
+
+The `lsst-versions`_ package is used to work around this restriction by determining version numbers based on the most recent formal version and the current weekly tag.
+For example, in ``daf_butler`` the version that was uploaded to `PyPI`_ for tag ``w.2023.42`` was ``26.2023.4200`` (where v26 was the most recent formal release tag at the time).
+The trailing ``00`` in the version number is used to allow different version numbers to be determined on ticket branches whilst developing the package between weekly tags and is the number of commits since the most recent weekly.
+
+Project Metadata
+----------------
+
+The ``project`` section is used to specify the core metadata for the package.
+For example, this is the content for the ``lsst-resources`` package:
+
+.. code-block:: toml
+
+    [project]
+    name = "lsst-resources"
+    description = "An abstraction layer for reading and writing from URI file resources."
+    license = {text = "BSD 3-Clause License"}
+    readme = "README.md"
+    authors = [
+        {name="Rubin Observatory Data Management", email="dm-admin@lists.lsst.org"},
+    ]
+    classifiers = [
+        "Intended Audience :: Developers",
+        "License :: OSI Approved :: BSD License",
+        "Operating System :: OS Independent",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+    ]
+    keywords = ["lsst"]
+    dependencies = [
+        "lsst-utils",
+        "importlib_resources",
+    ]
+    dynamic = ["version"]
+    requires-python = ">=3.10.0"
+
+The Rubin DM convention is that science pipelines packages that are to be distributed on `PyPI`_ should include the ``lsst-`` prefix in the name.
+This differs from the EUPS naming convention where the ``lsst`` is implicit.
+For example, the ``daf_butler`` EUPS package has a python distribution name of ``lsst-daf-butler``.
+Middleware packages use a dual license for distribution, with the `PyPI`_ package declaring the BSD 3-clause license.
+Most science pipelines packages use the GPLv3 license and for those packages the ``pyproject.toml`` should use:
+
+.. code-block:: toml
+
+    license = {text = "GPLv3+ License"}
+    classifiers = [
+      "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+    ]
+
+Every Rubin DM science pipelines package should be owned by the special DM account and that should always be included in the ``authors`` section.
+Additional authors can be included if required.
+
+The ``dependencies`` section can only refer to packages that are available on `PyPI`_ since this is the section that will be read during ``pip install``.
+
+Setuptools Configuration
+------------------------
+
+For ``setuptools`` builds additional configuration is needed so that the python files and data files can be located.
```

Comment:
This begs the question as to which packages should use `setuptools`, which should not, and what the alternatives are.

Reply:
Maybe I rephrase to something like "In this documentation we will use `setuptools`  for the build" or something.